### DISCUSSION TITLE
Is there a usage where components themselves contain "include" filter tags?

### DISCUSSION THREAD

**Discussion by simkimsia** [torchbox/django-pattern-library#197] [Ideas]
I have a component that contains references to markup, css, js, and also CDN imports.

- markup - obvious
- css - for the styling
- js - for behavior

I will place them in separate templates and "call" them using "include"

```html
<!DOCTYPE html>
<html>
  <head>
      <title>Demo 03 Easy to Customize Example in Tagify using include</title>
      {% include "tagify/cdn/css-cdn.html" %}
      <style>
        {% include "tagify/03_easy_to_customize/tagify.css" %}
      </style>
  </head>
  <body>
    {% include "tagify/03_easy_to_customize/tagify.html" %}
    {% include "tagify/cdn/js-cdn.html" %}
    <script>
      window.addEventListener('load', function () {
        {% include "tagify/03_easy_to_customize/tagify.js" %}
      });
    </script>
  </body>
<html>
```

Looks like this 
![image](https://user-images.githubusercontent.com/245021/179391397-0748b6c9-ddb5-4557-a8e2-8fe7f036c52b.png)

Is this a common use case amongst your users?

Do you have intention to allow add "sub-tabs" under the "template source" tab?

i was thinking of 3 views within "template source":

1. the template source as it is
2. the full markup after all the includes are rendered
3. a 2 column layout where left handside looks like 1 but you can "select" specific "include" and the right hand side column will auto show the markup for the selected include

Sounds like overkill, I know. But it's very common for me to use a lot of includes to build my components.

## Update

I was looking through the code, so basically somewhere here at https://github.com/torchbox/django-pattern-library/blob/983ac1ce32b751f1cc42f264cbab28f8866fb530/pattern_library/views.py#L70, need a way to inspect to see if include tag was used in the template source and add more variables.

**Comment by thibaudcolas** [comment]
I think that would be pretty cool! We had discussed creating links between patterns in the past (#6), this feels like another neat way to do it.

**Reply by simkimsia** [reply]
can i try my hand at forking this and doing my version of teh inner sub-tab and try to expand the includes?

what is the place where the template for this is displayed?

Is it at `pattern_library/index.html`?

https://github.com/torchbox/django-pattern-library/blob/983ac1ce32b751f1cc42f264cbab28f8866fb530/pattern_library/views.py#L28
